lib/compat/wordpress-6.2/formatting.php

```diff
@@ -0,0 +1,115 @@
+<?php
+/**
+ * New WordPress Formatting API.
+ *
+ * @package WordPress
+ */
+
+if ( ! function_exists( 'wp_word_count' ) ) {
+	/**
+	 * Count words or characters in a provided text string.
+	 *
+	 * @since 6.2.0
+	 *
+	 * @param string $text  Text to count elements in.
+	 * @param string $type The type of count. Accepts 'words', 'characters_excluding_spaces', or 'characters_including_spaces'.
+	 * @param array  $settings {
+	 *     Optional. Array of arguments used to overrides for settings.
+	 *
+	 *     @type string $html_regexp                        Optional. Regular expression to find HTML elements.
+	 *     @type string $html_comment_regexp                Optional. Regular expression to find HTML comments.
+	 *     @type string $space_regexp                       Optional. Regular expression to find irregular space
+	 *                                                      characters.
+	 *     @type string $html_entity_regexp                 Optional. Regular expression to find HTML entities.
+	 *     @type string $connector_regexp                   Optional. Regular expression to find connectors that
+	 *                                                      split words.
+	 *     @type string $remove_regexp                      Optional. Regular expression to find remove unwanted
+	 *                                                      characters to reduce false-positives.
+	 *     @type string $astral_regexp                      Optional. Regular expression to find unwanted
+	 *                                                      characters when searching for non-words.
+	 *     @type string $words_regexp                       Optional. Regular expression to find words by spaces.
+	 *     @type string $characters_excluding_spaces_regexp Optional. Regular expression to find characters which
+	 *                                                      are non-spaces.
+	 *     @type string $characters_including_spaces_regexp Optional. Regular expression to find characters
+	 *                                                      including spaces.
+	 *     @type array  $shortcodes                         Optional. Array of shortcodes that should be removed
+	 *                                                      from the text.
+	 * }
+	 * @return int The word or character count.
+	 */
+	function wp_word_count( $text, $type, $settings = array() ) {
```

Comment:
This function will be implemented in the core, but is this the correct directory to place the function?

Reply:
If you place it here it'll be backported when WordPress 6.2 comes around. The function should have the `gutenberg_` prefix while it lives in the plugin though and only be changed to `wordpress_` when it's moved to core. The backports happen manually and the release lead for 6.2 will check this folder. Here's the issue where backports for 6.1 were coordinated as an example - https://github.com/WordPress/gutenberg/issues/43440.

Alternatively you can open up a trac ticket and try to get it merged to core separately, but there's no real advantage to doing that. The only advantage would be if a core feature needs to use this function between now and 6.2.

Reply:
According to[ this comment](https://github.com/WordPress/gutenberg/pull/43403#discussion_r993617934), it appears that functions merged into core should have a wp_prefix, correct?

Reply:
I don't think that's right (cc @anton-vlasenko). If you check in the `lib/wordpress-6.1` folder for example there are no `wp` prefixed functions, only `gutenberg` prefixed functions.

This code will be in the Gutenberg plugin for a few months before being merged into core around the time of the 6.2 beta. During the time it's in the plugin it needs a `gutenberg` prefix as the `wp` prefix is reserved for code in the WordPress core codebase.

When the 6.2 beta comes around, the function will need to be implemented in the WordPress core codebase. The WP core implementation will then have the `wp` prefix.

Reply:
In the `lib/compat` directory, there are several functions with the wp prefix.
The wp_word_count function in this case might be used for the same purpose?

- `lib/compat/wordpress-6.0/functions.php`: [wp_recursive_ksort](https://github.com/WordPress/gutenberg/blob/3c9a0207f51d731ff4d8cfbfa69daa16f797639f/lib/compat/wordpress-6.0/functions.php#L19)
- `lib/compat/wordpress-6.1/theme.php`: [wp_theme_get_element_class_name](https://github.com/WordPress/gutenberg/blob/3c9a0207f51d731ff4d8cfbfa69daa16f797639f/lib/compat/wordpress-6.1/theme.php#L44)
- `lib/compat/wordpress-6.2/get-global-styles-and-settings.php`: [wp_theme_has_theme_json](https://github.com/WordPress/gutenberg/blob/3c9a0207f51d731ff4d8cfbfa69daa16f797639f/lib/compat/wordpress-6.2/get-global-styles-and-settings.php#L16), [wp_theme_clean_theme_json_cached_data](https://github.com/WordPress/gutenberg/blob/3c9a0207f51d731ff4d8cfbfa69daa16f797639f/lib/compat/wordpress-6.2/get-global-styles-and-settings.php#L43)

Reply:
That's confusing to me. I always thought `wp` was reserved for core (Gutenberg isn't part of core).

I'll ask around.

Reply:
In most cases I think @talldan is right. When we introduce a new function we need to `gutenberg` prefix it. When it lands in core we rename it(in core) to `wp` prefix. 

In this case I think it's okay, because it's used inside a `block/index.php` file and if I'm not wrong the functions there are stripped of `gutenberg` prefix for core.

Reply:
> In this case I think it's okay, because it's used inside a block/index.php file and if I'm not wrong the functions there are stripped of gutenberg prefix for core.

Ah yes, that might be why. 💡